### PR TITLE
Hosting Command Palette: Trim double quotes from selected items if present

### PR DESCRIPTION
**Pull request by katinthehatsite**
Related to https://github.com/Automattic/dotcom-forge/issues/4550

## Proposed Changes

This PR trims the double quotes from the selected site title before it is processed by the `document.querySelector`. This ensures that the command palette does not break if the site titles include double quotes.

This is what the error looks like if there are double quotes present before they are being processed by the `document.querySelector` in the command palette: 

Screenshot 2023-11-21 at 3 16 33 PM

## Testing Instructions

* Ensure that you have an Atomic site
* Set the site title under **Settings > General** on an Atomic site to include double quotes e.g. "My Site"
* First, test on `https://wpcalypso.wordpress.com/sites` by triggering `cmd + k` on Mac and `ctrl + k` on Windows
* Navigate to one of the SSH commands e.g. `Open SSH details`
* Try navigating to the site with double quotes in the site title
* Confirm this breaks the command palette
* Next, navigate to Calypso Live Link generated in this PR
* Trigger command palette with `cmd + k` on Mac and `ctrl + k` on Windows
* Try navigating to the site with double quotes in the site title
* Confirm that the command executes as expected 

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by sejas — APPROVED** [review]
Wow!, great catch @katinthehatsite . 🙌 

This PR solves the issue that I was able to replicate on production. I also tried adding other characters besides the quote, like `[`, `]`, but they worked fine.

We should upstream this change.
https://github.com/WordPress/gutenberg/blob/trunk/packages/commands/src/components/command-menu.js#L161C17-L164

I confirm that having a site with quotes or other weird characters don't break the command palette. And I was able to navigate to the SSH details. 👍 

# Screencast

https://github.com/Automattic/wp-calypso/assets/779993/8cb44699-126b-4e85-811a-4b3e0da3de83

File: client/components/command-pallette/wpcom-command-pallette.tsx [hunk 1]
```diff
@@ -69,16 +69,23 @@ interface CommandInputProps {
 function CommandInput( { isOpen, search, setSearch }: CommandInputProps ) {
 	const commandMenuInput = useRef< HTMLInputElement >( null );
 	const _value = useCommandState( ( state ) => state.value );
+	const sanitizedValue = useMemo( () => {
+		const removedQuotesValue = _value.replace( /"/g, '' ); // Remove double quotes from any selected items before processing input
```
Comment (danielbachhuber) [inline client/components/command-pallette/wpcom-command-pallette.tsx]: Should we use `escapeAttribute` from `@wordpress/escape-html`? https://developer.wordpress.org/block-editor/reference-guides/packages/packages-escape-html/#escapeattribute

Or how do we match the sanitization the library is already applying, to make sure our value exactly matches what's rendered on the page?

Also, what other characters might cause problems?

Comment (sejas) [reply client/components/command-pallette/wpcom-command-pallette.tsx]: > Also, what other characters might cause problems?

Yesterday reviewing the PR I found that the double quotes character is problematic. The rest worked fine.

**Comment by sejas** [comment]
@katinthehatsite, let's pair on this.
All this logic is for accessibility improvements and should be implemented directly on `cmdk`.  I think we can get the id withtout accessing the DOM elements at all.

**Comment by sejas** [comment]
For context:
The accessibility issue was fixed on `cmdk` but there hasn't been a new release since March. Therefore those have the same risk.

- https://github.com/pacocoursey/cmdk/pull/108
- https://github.com/WordPress/gutenberg/pull/52930

**Comment by sejas** [comment]
Hey Kat, I created this PR using yours as a base. We are taking a different/better approach from `cmdk` and `@wordpress/commands` https://github.com/Automattic/wp-calypso/pull/84412

File: client/components/command-pallette/wpcom-command-pallette.tsx [hunk 2]
```diff
@@ -34,12 +35,13 @@ export function CommandMenuGroup( {
...
 			{ commands.map( ( command ) => {
+				const itemValue = command.searchLabel ?? command.label;
 				return (
 					<Command.Item
 						key={ command.name }
-						value={ command.searchLabel ?? command.label }
+						value={ itemValue }
 						onSelect={ () => command.callback( { close, setSearch } ) }
-						id={ command.name }
+						id={ cleanForSlug( itemValue ) }
```
Comment (katinthehatsite) [inline client/components/command-pallette/wpcom-command-pallette.tsx:44]: I like this solution with `cleanForSlug`! TIL: https://developer.wordpress.org/block-editor/reference-guides/packages/packages-url/#cleanforslug

**Comment by katinthehatsite** [comment]
@sejas - thanks for all the research and the refactor. I think if we can make the fix in Gutenberg, this is definitely the approach that we should be taking!

**Pull request merged**